content/PDOStatementExecute/PDOStatementExecute.q.md

```diff
@@ -0,0 +1,1600 @@
+# Prepared Statement with Execute
+
++++
+
+### Part 1: Sample Code Analysis
+
+:::
+
+/// type=REPL, readonly=true, filename=[connection.php,student.php]
+
+```php
+// connection.php
+<?php
+    $host = 'localhost';
+    $db = 'LibraryDB';
+    $port = '5432';
+    $username = 'postgres';
+    $password = 'Admin01';
+
+    try {
+        $conn = new PDO("pgsql:host=$host;port=$port;dbname=$db;user=$username;password=$password");
+        if ($conn) {
+            echo "Successfully connected to the database. <br />";
+        }
+    } catch (Exception $e) {
+        echo "Unable to establish a connection."; 
+    }
+?>
+```
+
+```php
+// student.php
+<?php
+    require_once("connection.php");
+
+    $data = array(
+        'min_birth_date' => '1999-01-01',
+        'max_birth_date' => '2000-01-01'
+    );
+    
+    $sql = "SELECT * FROM students WHERE birth_date >= :min_birth_date AND birth_date < :max_birth_date";
+
+    $pstmt = $conn->prepare($sql);
+    $pstmt->bindParam(':min_birth_date', $data['min_birth_date'], PDO::PARAM_STR);
+    $pstmt->bindParam(':max_birth_date', $data['max_birth_date'], PDO::PARAM_STR);
+    $pstmt->execute();
+
+    $data = $pstmt->fetchAll();
+
+    foreach ($data as $row) {
+        echo $row['student_id'].$row['first_name'].$row['last_name'].$row['birth_date'].$row['gender']. ' ';
+    }
+?>
+```
+/// type=SS, answer=[5] 
+
+Execute the program. What is its output?
+
+- It produces an error.
+
+- No output is displayed.
+
+- It prints `Unable to establish a connection.`.
+
+- It prints `Successfully connected to the database.` and `7CarloPears1998-04-04M 8GenevieLieser1996-05-25F`.
+
+- It prints `Successfully connected to the database.` and `1JohnSmith1999-02-10M 2SamanthaDanes1999-10-12F 6AlisaElls1999-06-30F`.
+
+
+/// type=MS, answer=[2,3]
+
+Which of the following are array values in the associative array `$data`?
+
+- `students`
+
+- `1999-01-01`
+
+- `2000-01-01`
+
+- `min_birth_date`
+
+- `max_birth_date`
+
+
+/// type=MS, answer=[2,3,5]
+
+Which statements correctly describe the code on lines 4, 5, 6, and 7 of `student.php`?
+
+- It displays the elements of the associative array `$data`.
+
+- It holds the elements of the associative array `$data` using the `array()` construct.
```

Comment:
This is incorrect... Please check associative array lesson in PHP intermediate course. Thanks.